oauth2_provider/oauth2_validators.py

```diff
@@ -541,3 +581,111 @@ def validate_refresh_token(self, refresh_token, client, request, *args, **kwargs
 
         except RefreshToken.DoesNotExist:
             return False
+
+    @transaction.atomic
+    def _save_id_token(self, token, request, expires, *args, **kwargs):
+
+        scopes = request.scope or " ".join(request.scopes)
+
+        if request.grant_type == "client_credentials":
+            request.user = None
+
+        id_token = IDToken.objects.create(
+            user=request.user,
+            scope=scopes,
+            expires=expires,
+            token=token.serialize(),
+            application=request.client,
+        )
+        return id_token
+
+    def get_jwt_bearer_token(self, token, token_handler, request):
+        return self.get_id_token(token, token_handler, request)
+
+    def get_id_token(self, token, token_handler, request):
+
+        key = jwk.JWK.from_pem(oauth2_settings.OIDC_RSA_PRIVATE_KEY.encode("utf8"))
+
+        # TODO: http://openid.net/specs/openid-connect-core-1_0.html#HybridIDToken2
+        # Save the id_token on database bound to code when the request come to
+        # Authorization Endpoint and return the same one when request come to
+        # Token Endpoint
+
+        # TODO: Check if at this point this request parameters are alredy validated
+
+        expiration_time = timezone.now() + timedelta(seconds=oauth2_settings.ID_TOKEN_EXPIRE_SECONDS)
+        # Required ID Token claims
+        claims = {
+            "iss": 'https://id.olist.com',   # HTTPS URL
```

Comment:
that's right ? maybe this value could come from the constructor or something more generic